File: Iteration2/code/CSE3063GRP12/src/core/database/concretes/JsonManager.java
package core.database.concretes;

import com.google.gson.Gson;

import java.io.FileReader;
import java.io.FileWriter;
import core.database.abstracts.DatabaseManager;

public class JsonManager extends DatabaseManager {
    private Gson gson;

    public JsonManager() {

        gson = new Gson();
    }

    public <T> T read(String path, Class<T> classOfT) {
        try (FileReader reader = new FileReader(path)) {
            var result = gson.fromJson(reader, classOfT);
            return result;
        } catch (Exception e) {
            // e.printStackTrace();
            System.err.println(e.getMessage());
            return null;
        }
    }

    public <T> void write(String path, T object) {
        String json = gson.toJson(object);
        try (FileWriter writer = new FileWriter(path)) {
            writer.write(json);
        } catch (Exception e) {
            // e.printStackTrace();
        }
    }
}

File: Iteration3/code/CSE3063GRP12/src/core/database/concretes/JsonManager.py
import json
from core.database.abstracts.DatabaseManager import DatabaseManager

class JsonManager(DatabaseManager):
    def read(self, path, class_of_t):
        try:
            with open(path, 'r') as file:
                data = json.load(file)
                return class_of_t(**data) 
        except Exception as e:
            return None

    def write(self, path, object):
        try:
            with open(path, 'w') as file:
                if hasattr(object, 'to_dict'):
                        data = object.to_dict()  # Use the to_dict() method if available
                else:
                    data = object.__dict__
                json.dump(data, file)
        except Exception as e:
            pass